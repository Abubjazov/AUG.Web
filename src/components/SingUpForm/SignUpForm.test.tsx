import { fireEvent, render, screen } from '@testing-library/react'
import MockedProvider from 'mockData/mockedReduxProvider'

import SignUpForm from './SignUpForm'

describe('SignUpForm', () => {
  test('should render SignUpForm default', () => {
    const mockFn = vi.fn()

    const { asFragment } = render(
      <MockedProvider>
        <SignUpForm userFunction={mockFn} onSignUp={mockFn} />
      </MockedProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call SignUn function', () => {
    const mockFn = vi.fn()

    render(
      <MockedProvider>
        <SignUpForm userFunction={mockFn} onSignUp={mockFn} />
      </MockedProvider>,
    )

    fireEvent.click(screen.getByText('Sign up'))

    expect(mockFn).toHaveBeenCalledTimes(0)

    fireEvent.change(screen.getByPlaceholderText('email@example.com'), {
      target: { value: 'email@test.tst' },
    })

    fireEvent.change(screen.getByPlaceholderText('password'), {
      target: { value: '12345678' },
    })

    fireEvent.change(screen.getByPlaceholderText('password confirm'), {
      target: { value: '12345678' },
    })

    fireEvent.click(screen.getByText('Sign up'))

    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
