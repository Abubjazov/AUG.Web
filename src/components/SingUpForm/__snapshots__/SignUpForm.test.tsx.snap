// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignUpForm > should render SignUpForm default 1`] = `
<DocumentFragment>
  <div
    class="_root_b75372"
  >
    <span
      class="_title_b75372"
    >
      Sign 
      <span
        class="_red-text_b75372"
      >
        up
      </span>
    </span>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="email-input"
        name="email"
        placeholder="email@example.com"
        type="email"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="password-input"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="_root_672eb8"
    >
      <input
        data-testid="confirm-password-input"
        name="confirm-password"
        placeholder="password confirm"
        type="password"
        value=""
      />
    </div>
    <div
      class="_buttons_b75372"
    >
      <button
        class="_root_91d6ec _contained-red_91d6ec  _button_b75372"
        data-testid="base-button"
        type="button"
      >
        Sign up
      </button>
      <button
        class="_root_91d6ec _outlined-white_91d6ec  _button_b75372"
        data-testid="base-button"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
